lib/cocoapods/generator/module_map.rb

```diff
@@ -35,14 +40,24 @@ def save_as(path)
       # @return [String]
       #
       def generate
-        <<-eos.strip_heredoc
+        result = <<-eos.strip_heredoc
           framework module #{target.product_module_name} {
             umbrella header "#{target.umbrella_header_path.basename}"
 
             export *
             module * { export * }
-          }
         eos
+
+        result << "\n#{generate_private_header_exports}" unless private_headers.empty?
+        result << "}\n"
+      end
+
+      private
+
+      def generate_private_header_exports
+        private_headers.reduce('') do |string, header|
+          string << %(  private header "#{header}\n")
+        end
```

Comment:
There doesn't seem to be any tests to ensure `generate` works, can we add some? (I understand if you feel this is outside of this PR, can be done separately, but I think it makes sense to add them in this PR to make sure this works correctly).